app/services/course/assessment/question/programming_codaveri_service.rb

```diff
@@ -62,15 +62,18 @@ def initialize(question, attachment)
   # Constructs codaveri question problem object and send an API request to Codaveri to create/update the question.
   #
   # @param [Course::Assessment::ProgrammingPackage] package The programming package attached to the question.
-  def create_from_package(package)
+  def create_or_update_from_package(package)
     construct_problem_object(package)
-    create_codaveri_problem
+
+    @is_update_problem ? update_codaveri_problem : create_codaveri_problem
```

Comment:
Check the update condition? What about sync status?
`@is_update_problem = @question.codaveri_id.present?`

Reply:
I added the gatekeeping inside `update_codaveri_problem`, only to proceed if `is_synced_with_codaveri` is false